www/viewgame.js

```diff
@@ -0,0 +1,317 @@
+'use strict';
+
+function trim(str) {
+    return str.replace(/^\s+|\s+$/g, '');
+}
+
+function tagSorter(a, b)
+{
+    a = a.tag.toLowerCase();
+    b = b.tag.toLowerCase();
+    return (a > b ? 1 : a < b ? -1 : 0);
+}
+
+class TagTable {
+    constructor(gameid, tagList) {
+        this.gameid = gameid;
+        this.dbTagList = tagList;
+        this.memTagList = [];
+    }
+
+    addEventListeners(isAdmin) {
+        const buttons = [
+            ['#myTagList_edit', () => {this.editTags();}],
+            ['.viewgame__tagEditorContainer .viewgame__cancel a', () => {this.closeTags();}],
+            ['form[data-tag-button="add"]', () => {this.addTags();}, 'submit'],
+            ['#viewgame-add-tags-button', () => {this.addTags();}],
+            ['#viewgame-save-tags-button', () => {this.saveTags();}],
+            ['#viewgame-save-tags-button-delete', () => {this.saveTagsDelete();}],
+            ['#tagDeletor .viewgame__cancel a', () => {this.closeTags('tagDeletor');}],
+        ];
+        if (isAdmin) {
+            buttons.push(['#myTagList_delete', () => {this.deleteTags();}]);
+        }
+        for (const [selector, handler, listener] of buttons) {
+            document.querySelectorAll(selector).forEach(e => {
+                e.addEventListener(listener ? listener : 'click', (ev) => {
+                    ev.preventDefault();
+                    handler();
+                });
+            });
+        }
+    }
+
+    doTagCkBox(id, stat)
+    {
+        var i = parseInt(id.substr(5));
+        this.memTagList[i].tagcnt += (stat ? 1 : -1);
+        this.memTagList[i].gamecnt += (this.memTagList[i].isNew ? (stat ? 1 : -1) : 0);
+        this.memTagList[i].isMine = stat ? 1 : 0;
+        var ce = document.getElementById("tagCnt" + i);
+        var ct = this.memTagList[i].tagcnt;
+        ce.innerHTML = (this.memTagList[i].isNew || ct < 2  ? "" :
+                        "(x" + this.memTagList[i].tagcnt + ")");
+        ce.title = ct + " member" + (ct > 1 ? "s have" : " has")
+                   + " tagged this game with \"" + this.memTagList[i].tag + "\"";
+    }
+
+
+    dispTagTable(tableID, lst, editor, deleteTags)
+    {
+        var tbl = document.getElementById(tableID);
+        tbl.innerHTML = "";
+        var row = null, rownum = 0;
+        for (var i = 0 ; i < lst.length ; i++) {
+            var t = encodeHTML(lst[i].tag);
+            var u = encodeURI8859(lst[i].tag);
+            var ct = lst[i].tagcnt;
+            var cg = lst[i].gamecnt;
+            var m = lst[i].isMine;
+            var n = lst[i].isNew;
```

Comment:
All these single-letter variable names make this code hard to read (I'll merge it anyway, though)